scripts/Test-Signatures.ps1

```diff
@@ -0,0 +1,209 @@
+[CmdletBinding()]
+param (
+    [Parameter()]
+    [String]
+    $Path = "$PSScriptRoot\packages",
+
+    [Parameter()]
+    [Switch]
+    $CIBuild
+)
+
+Add-Type -AssemblyName System.IO.Compression.FileSystem
+
+# Allowed certificate subjects
+$global:ValidSubjects = @( `
+    "CN=Microsoft Corporation, O=Microsoft Corporation, L=Redmond, S=Washington, C=US", `
+    "CN=Microsoft Corporation, OU=MOPR, O=Microsoft Corporation, L=Redmond, S=Washington, C=US" `
+);
+
+# Allowed certificate thumbprints
+$global:ValidThumbprints = @( `
+    "B9EAA034C821C159B05D3521BCF7FEB796EBD6FF", `
+    "98ED99A67886D020C564923B7DF25E9AC019DF26", `
+    "5EAD300DC7E4D637948ECB0ED829A072BD152E17", `
+    "67B1757863E3EFF760EA9EBB02849AF07D3A8080", `
+    "9DC17888B5CFAD98B3CB35C1994E96227F061675", `
+    "62009AAABDAE749FD47D19150958329BF6FF4B34", `
+    "3BDA323E552DB1FDE5F4FBEE75D6D5B2B187EEDC" `
+)
+
+# Allow-list for unsigned binaries
+$global:UnsignedAllowList = @{
+    # "PackageId" = relative paths
+    "Microsoft.TestPlatform" = @( `
+        ".\tools\net451\Common7\IDE\Extensions\TestPlatform\Newtonsoft.Json.dll" `
+    )
+
+    "Microsoft.TestPlatform.CLI" = @( `
+        ".\contentFiles\any\netcoreapp2.1\TestHost\Newtonsoft.Json.dll", `
+        ".\contentFiles\any\netcoreapp2.1\Newtonsoft.Json.dll"
+    )
+    
+    "Microsoft.TestPlatform.Portable" = @( `
+        ".\tools\net451\Newtonsoft.Json.dll", `
+        ".\tools\netcoreapp2.1\TestHost\Newtonsoft.Json.dll", `
+        ".\tools\netcoreapp2.1\Newtonsoft.Json.dll"
+    )
+}
+
+
+$global:Depth = 0;
+
+function Get-RelativePath($path, $relativeTo) {
+    Push-Location
+    Set-Location $relativeTo
+    $path = Resolve-Path $path -Relative
+    Pop-Location
+    
+    return $path
+}
+
+function Push-Depth {
+    $global:Depth = $global:Depth + 1
+}
+
+function Pop-Depth {
+    $global:Depth = $global:Depth - 1
+
+    if($global:Depth -lt 0) {
+        $global:Depth = 0;
+    }
+}
+
+function Clear-Line($message = "") {
+    if(-not $CIBuild) {
+        $x = [System.Console]::get_CursorLeft()
+        Write-Host "`r$(' ' * $x)$message`r" -NoNewline
+    }
+}
+
+function Update-Operation($operation, $message, [switch]$NewLine, $suppress=$false) {
+    if($suppress) {
+        return;
+    }
+    
+    Clear-Line;
+    Write-Host "$('  ' * $Depth)[$operation] " -ForegroundColor Yellow -NoNewline

```

Comment:
Rather than inventing you own reporting use Write-Progress, or write Write-Verbose.

Reply:
I wanted to have a cleaner output. (Not hundreds of assemblies listed in the output unless needed) I'll try to switch that.